### ISSUE TITLE
Environment for future_lapply 

### ISSUE THREAD

**Issue by dipterix** [HenrikBengtsson/future.apply#70]
In `future`, we can set environment where to search for globals, but in `future_lapply`, this option seems to be removed.

Is it possible to do this:

([original code](https://github.com/HenrikBengtsson/future.apply/blob/461a3e2ef99acddf92a4b614b9d828579912d29d/R/future_lapply.R#L188))
``` r
future_lapply <- function(X, FUN, ..., future.envir2 = parent.frame()){
  ...
  values <- future_xapply(
    ...,
    future.envir2 = future.envir2
  )
}

```

In this way, `future_xapply` will be able to receive environment in case other Future classes need to process the environment.

https://github.com/HenrikBengtsson/future.apply/blob/fcec8077a4eb8da0b4e60dd67a20fca33715942f/R/future_xapply.R#L186-L195

We only need to change line 188 to

``` r
fs[[ii]] <- future(
      ...
      envir = future.envir2,
      ...
)
```

**Comment by HenrikBengtsson** [comment]
Yes, this is a known issue and on the road map (for way too long), e.g. Issue #62 and https://github.com/HenrikBengtsson/future.apply/issues/64#issuecomment-733487812.  I just need some deep focus to get this correct; passing down `envir = parent.frame()` as-is will result in some revdep complaints related to globals and `...` that need to be understood and sorted out.

**Comment by HenrikBengtsson** [comment]
Also, I was hoping to have gotten around and created https://github.com/HenrikBengtsson/future.mapreduce by now so it can just implemented it there and then future.apply, doFuture, and furrr could depend on that.

**Comment by dipterix** [comment]
Sounds good!

**Comment by beauchamplab** [comment]
Thanks for considering the fix and update me with your thought. Really appreciate it.

**Comment by dipterix** [comment]
Thanks for considering the fix and update me with your thought. Really appreciate it.
